### DISCUSSION TITLE
Inquiry about Collecting Kubernetes Cluster Logs with Grafana Agent Deployed via Helm

### DISCUSSION THREAD

**Discussion by monlor** [grafana-cold-storage/agent#4594] [Q&A] [answered]
Issue Description:
I have successfully deployed Grafana Loki in my Kubernetes cluster using Helm, and as part of the deployment, I included Grafana Agent to collect logs from Loki itself. Currently, Grafana Agent is configured to scrape logs from its own components and send them to Loki for storage and analysis. This setup allows me to collect logs related to Grafana Agent and Loki.

Use Case:
Now, I want to extend the log collection capability to cover the entire Kubernetes cluster, including logs from all the pods running within it. My objective is to have a comprehensive log collection system that includes both Loki's internal logs and logs from the entire Kubernetes environment.

Question:
Given that I have already deployed Grafana Agent via Helm and it is successfully collecting logs from its own components, is it necessary to additionally deploy Promtail to achieve comprehensive log collection from the entire Kubernetes cluster?

**Comment by tpaschalis** [comment, marked as answer]
To achieve this, you can deploy Grafana Agent in Daemonset mode, mount the /var/log/pods or equivalent directory with the correct permissions and scrape logs from there.

If you're feeling adventurous, in Flow mode we have a new experimental component, [loki.source.kubernetes](https://grafana.com/docs/agent/next/flow/reference/components/loki.source.kubernetes/) that reads logs directly from the Kubernetes API and doesn't require a Daemonset to work, but might not be suitable for production workloads yet, YMMV.
